src/test/java/com/cag/cagbackendapi/controller/UserControllerTest.kt

```diff
@@ -0,0 +1,58 @@
+package com.cag.cagbackendapi.controller
+
+import com.cag.cagbackendapi.constants.DetailedErrorMessages
+import com.cag.cagbackendapi.error.exceptions.UnauthorizedException
+import com.cag.cagbackendapi.model.UserModel
+import com.cag.cagbackendapi.service.user.UserService
+import com.cag.cagbackendapi.service.validation.ValidationService
+import com.nhaarman.mockito_kotlin.*
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.InjectMocks
+import org.mockito.junit.jupiter.MockitoExtension
+
+@ExtendWith(MockitoExtension::class)
+class UserControllerTest {
+
+    private var userService: UserService = mock()
+    private var validationService: ValidationService = mock()
+
+    @InjectMocks
+    private lateinit var userController: UserController
+
+    @Test
+    fun registerUser_validInput_returns201() {
+        val testAuthKey = "testAuthKey"
+        val testUser = UserModel(name = "John Smith", email = "johnjohn@aol.com")
+
+        doNothing().whenever(validationService).validateAuthKey(testAuthKey)
```

Comment:
Why do you have it doing nothing? Is it because it is a test?

Reply:
doNothing() means that the function in question will not return a value. The doNothing() is still important because it checks that the function was called correctly and only a certain amount of times. Because of this "doNothing().whenever(validationService).validateAuthKey(testAuthKey)" - if the validationService.validateAuthKey() method isn't called, the test will fail. If the function is called twice instead of once, the test will also fail. If the function is called with the wrong testAuthKey value, the function will also fail. So it does a lot of important checks!